libimage/filters.go

```diff
@@ -268,8 +268,15 @@ func filterManifest(ctx context.Context, value bool) filterFunc {
 }
 
 // filterReferences creates a reference filter for matching the specified value.
-func filterReferences(value string) filterFunc {
+func filterReferences(r *Runtime, value string) filterFunc {
+	lookedUp, _, _ := r.LookupImage(value, nil)
 	return func(img *Image) (bool, error) {
+		if lookedUp != nil {
+			if lookedUp.ID() == img.ID() {
```

Comment:
This seems imprecise to me.

The code below seems fairly clear that `fedora:38` is intended to match `example.com/any/namespaces/fedora:38`; and even that wildcards work, so `fe*ra:38` is intended to match.

Is that correct?

If so, `fe*ra@sha256:$digest` should also match — but `LookupImage` can (I hope?) never resolve `fe*ra`.

IOW, if the underlying problem is that `img.NamesReferences()` does not contain digests, I think that needs to be solved directly here, not by relying on the `LookupImage` crutch (and on the `lookupImageInDigestsAndRepoTags` code path for matching short names) — maybe by combining `img.NamesReference()` with  `img.RepoDigests()` when building `candidates`.

Now, `RepoDigests` is imprecise and relying on it would _also_ introduce a weak variant of #1248 (where an image that exists on registry as repo1@digest1 and repo2@digest2 can be referenced using repo1@digest2, although that exists on no registry — but it’s a weaker variant because otherRepo@digest1 does not match). Ideally we should fix `RepoDigests`, but that’s a whole other set of PRs again.

Reply:
Even more extreme, wildcards inside the digest? `fedora@sha*:a?b*c` ?